frontend/src/static/js/v_summary.js

```diff
@@ -282,6 +302,43 @@ window.vSummary = {
       this.getDates();
       this.sortFiltered();
     },
+    processFileTypes() {
+      const selectedColors = ['#e6194b', '#3cb44b', '#ffe119', '#4363d8', '#f58231', '#911eb4', '#46f0f0', '#f032e6',
+        '#bcf60c', '#fabebe', '#008080', '#e6beff', '#9a6324', '#fffac8', '#800000', '#aaffc3', '#808000', '#ffd8b1',
+        '#000075', '#808080'];
+
+      this.repos.forEach((repo) => {
+        const user = repo.users[0];
+        const colors = {};
+        let i = 0;
+        if (this.isDefaultFileFormats(Object.keys(user.fileTypeContribution))) {
+          user.fileTypeContribution = this.sortKeysAlphabetically(user.fileTypeContribution);
```

Comment:
What I actually meant is, why don't we just sort it regardless of whether is it default file formats?
Making a const `defaultFileFormats` is not really useful because we will have to keep PR which modifies the default file formats in check so that the devs who change the java file will remember to change for the js too.

Reply:
By right, there shouldn't be a need to sort the default file formats in the first place as the json would outputting the java objects in the same order every time. We would have problem passing the test cases if it didn't.

Reply:
> What I actually meant is, why don't we just sort it regardless of whether is it default file formats?

This was actually my original implementation, which was modified after my discussion with @damithc. We agreed that if the user specifies the file formats, then the contribution bar and legend should reflect those file formats in order specified by the user, and not alphabetically. If no file format is specified by the user, then all default file formats should be shown in the legend in alphabetical order. Since information about the default file formats is not passed from the backend to frontend, i need to create a const ```defaultFileFormats``` for the sorting.

Reply:
> By right, there shouldn't be a need to sort the default file formats in the first place as the json would outputting the java objects in the same order every time. We would have problem passing the test cases if it didn't.

What do you mean? I am using a ```HashMap``` in the backend which doesn't maintain any ordering of the keys, it just adds/increments file type count as encountered in ```FileResultAggregator```. Since the ordering is alphabetical only for ```defaultFileFormat```, there is a need to sort it in the frontend.

Reply:
>What do you mean? I am using a `HashMap` in the backend which doesn't maintain any ordering of the keys, it just adds/increments file type count as encountered in `FileResultAggregator`. Since the ordering is alphabetical only for `defaultFileFormat`, there is a need to sort it in the frontend.

In that case, how would you be sure that the legend will always `reflect those file formats in order specified by the user`?

Reply:
I have changed the implementation to use a ```LinkedHashMap``` which uses the ordering of ```DEFAULT_FORMATS``` specified in the backend which is already sorted alphabetically, and uses the file formats in order specified by user, therefore I've removed the sorting in the frontend.